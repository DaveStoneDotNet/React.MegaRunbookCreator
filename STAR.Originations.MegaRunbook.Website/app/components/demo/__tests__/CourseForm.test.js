import React       from 'react';
import { mount }   from 'enzyme';
import { shallow } from 'enzyme';
import renderer    from 'react-test-renderer';

import CourseForm  from '../CourseForm';

function getTestProps(isSaving) {
    return {
        course:     {}, 
        allAuthors: [], 
        onSave:     () => {},
        onChange:   () => {}, 
        isSaving:   isSaving, 
        errors:     {}
    };
}

function getShallowCourseFormWrapper(isSaving) {
    const props = getTestProps(isSaving);
    return shallow(<CourseForm {...props} />);
}

describe('CourseForm via Enzyme', () => {
    it('renders form and h1', () => {

        const shallowCourseFormWrapper = getShallowCourseFormWrapper(false);
        expect(shallowCourseFormWrapper.find('form').length).toBe(1);
        expect(shallowCourseFormWrapper.find('h1').text()).toEqual('Manage Course');
    });

    it('save button is labeled "Save" when not saving', () => {
        const shallowCourseFormWrapper = getShallowCourseFormWrapper(false);
        expect(shallowCourseFormWrapper.find('input').props().value).toBe('Save');
    });

    it('save button is labeled "Saving..." when saving', () => {
        const shallowCourseFormWrapper = getShallowCourseFormWrapper(true);
        expect(shallowCourseFormWrapper.find('input').props().value).toBe('Saving...');
    });

    it('rendered CourseForm matches snapshot', () => {
        const props = getTestProps(false);
        
        const tree = renderer.create(<CourseForm {...props} />).toJSON();
        // console.log(tree);

        // If the test fails (but is valid), press 'u' to update the snapshot...
        expect(tree).toMatchSnapshot();

        // Unnecessary - Illustrative only. Does the same thing as 'toMatchSnapshot'... 
        // ... just more explicitly and granularly.
        
        expect(tree.children.length).toBe(6);
        expect(tree.children[0].type).toBe('h1');
        expect(tree.children[5].type).toBe('input');
    });
});
