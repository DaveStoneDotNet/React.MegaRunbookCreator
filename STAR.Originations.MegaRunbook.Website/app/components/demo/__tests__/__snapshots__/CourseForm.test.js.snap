// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseForm via Enzyme rendered CourseForm matches snapshot 1`] = `
<form>
  <h1>
    Manage Course
  </h1>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="title"
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="authorId"
    >
      Author
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="authorId"
        onChange={[Function]}
        type="text"
        value={undefined}
      >
        <option
          value=""
        >
          Select Author
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="category"
    >
      Category
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="category"
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="length"
    >
      Length
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="length"
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <input
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="submit"
    value="Save"
  />
</form>
`;
